app/api/modules.py

```diff
@@ -1,11 +1,31 @@
-from flask_rest_jsonapi import ResourceDetail
+from flask_rest_jsonapi import ResourceDetail, ResourceList
 
 from app.api.bootstrap import api
 from app.api.schema.modules import ModuleSchema
 from app.models import db
 from app.models.module import Module
 
 
+class ModuleList(ResourceList):
+    """
+    module list
+    """
+    def before_get(self, args, kwargs):
+        """
+        before get method to get the resource id for fetching details
+        :param args:
+        :param kwargs:
+        :return:
+        """
+        kwargs['id'] = 1
+
+    decorators = (api.has_permission('is_super_admin', methods="POST", id="1"),)
```

Comment:
Why aren't we checking for super admin permissions for a ```GET``` request ? Not sure about it but I think we should also check for ```GET``` method. Please point out if I am wrong.

Reply:
> Because we require only one object in the system for Modules.

So that one object is accessible by anyone or just the super admin ?

Reply:
super admin requires this only.

Reply:
Yes, that's my question. Then we should check it for the ```GET``` request also. I think you missed it.